src/scp/NominationProtocol.cpp

```diff
@@ -664,4 +817,19 @@ NominationProtocol::getLatestMessage(NodeID const& id) const
     }
     return nullptr;
 }
+
+static const int NOMINATION_MAX_TIMEOUT_SECONDS = (30 * 60); //same value as MAX_TIMEOUT_SECONDS defined within SCPDriver.cpp
+std::chrono::milliseconds
+NominationProtocol::computeNominationTimeout(uint32 roundNumber)
+{
+    if (roundNumber > NOMINATION_MAX_TIMEOUT_SECONDS)
+    {
+        leaderNominationTimeoutInMilisec = std::chrono::duration_cast<std::chrono::milliseconds> (std::chrono::seconds(NOMINATION_MAX_TIMEOUT_SECONDS));
+    }
+    else
+    {
+        leaderNominationTimeoutInMilisec.operator+=(std::chrono::milliseconds(BASE_LEADER_NOMINATION_TIMEOUT_MS)); //increment by 250 ms for each round
```

Comment:
Why not just multiple round number by BASE_LEADER_NOMINATION_TIMEOUT_MS?

Reply:
Round number is not always accurate.  This value is a member of the NominationProtocol class and there are places that increments this value.  In my testing, when it goes beyond a certain round (e.g. 4th), the value is not accurate.

Reply:
I guess it doesn't matter too much as long as the timeout is being reset properly, though that seems annoying. As for accuracy, using round number is good enough for the original code.

Reply:
Is there a NominationProtocol per thing being nominated? or is NominationProtocol shared between all things needing to be nominated?

Reply:
NominationProtocol per consensus round.

Reply:
A NominationProtocol object should be used by more than just one consensus round right? Otherwise the leaderNominationTimeoutInMilisec would just be 250 for all rounds.

Reply:
Sorry, NominationProtocol  is instantiated for each closing of ledger.  Within closing of ledger, there can be 1:* nomination rounds.